gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleCloudStorageStatisticsTest.java

```diff
@@ -89,23 +95,28 @@ public void gcs_requestCounter() throws Exception {
   @Test
   public void gcs_rateLimitCounter() {
     // verify for http event i.e. via Apiary
-    GoogleCloudStorageEventBus.postOnHttpResponseStatus(429);
+    GoogleCloudStorageEventBus.postGcsJsonApiEvent(new TestGcsApiEvent(429));
     GhfsGlobalStorageStatistics verifyCounterStats = new GhfsGlobalStorageStatistics();
+    verifyCounterStats.incrementCounter(GCS_METADATA_REQUEST, 1);
     verifyCounterStats.incrementCounter(GCS_API_CLIENT_RATE_LIMIT_COUNT, 1);
     verifyCounterStats.incrementCounter(GCS_API_CLIENT_SIDE_ERROR_COUNT, 1);
     verifyStatistics(verifyCounterStats);
 
     storageStatistics.reset();
 
     // verify for gRPC event i.e. via java-storage
+    verifyCounterStats = new GhfsGlobalStorageStatistics();
```

Comment:
why we have to recreate again?

Reply:
In the case of gRPC GCS_METADATA_REQUEST will not be updated. Hence need this.